### PR TITLE
fix: adjust `doctor` command to work with custom `ios/` folder location

### PR DESCRIPTION
**Pull request by szymonrybczak**
Summary:
---------

Fixes https://github.com/react-native-community/cli/issues/1989
Right now when looking for `.xcode.env` file, we look in directories defined in `react-native.config.js`. It fixes the problem for the brownfield setups, where very often there `ios/` folder is not in the default location. 

Test Plan:
----------
1. Clone the repository and do all the required steps from the [Contributing guide](https://github.com/react-native-community/cli/blob/main/CONTRIBUTING.md)
2. Run this command:

```sh
node /path/to/react-native-cli/packages/cli/build/bin.js doctor
```
Should find the `.xcode.env`, and if it doesn't exist - after fixing issues by pressing `f` inside terminal, it should create the file in correct location.

Checklist
----------

- [x] Documentation is up to date to reflect these changes.
- [x] Follows commit message convention described in [CONTRIBUTING.md](https://github.com/react-native-community/cli/blob/main/CONTRIBUTING.md#commit-message-convention)

### CODE REVIEW

**Comment by szymonrybczak** [comment]
@sbalay Mind checking if it works in your custom setup? 🙏 Here's the [guide](https://github.com/react-native-community/cli/blob/main/CONTRIBUTING.md#testing-your-changes) for testing changes locally.

**Comment by sbalay** [comment]
Hey 👋 

Just tried it, seems to work fine!

Created a tiny PR to update the guide to test local changes that was missing cli-doctor in the linking step: https://github.com/react-native-community/cli/pull/2000

Another side note is that, while playing around, I did the following in a normal react-native project:

```
mv ios other-ios
yarn react-native doctor
```

The cli worked fine and detected that the environment is correctly set up. When adding more logs to cli-doctor to check what was the inferred ios project folder, it was correctly set to `other-ios`. However, I didn't change anything anywhere else (didn't update react-native.config.js either). How did the cli manage to infer the correct folder?

**Comment by szymonrybczak** [comment]
Hello 👋 Thanks for testing!
>  How did the cli manage to infer the correct folder?

So before running every command CLI is running `loadConfig()` function that resolves config, and for iOS CLI is not looking for `ios/` directory but is looking for `Podfile`, and then if the CLI found `Podfile` under some directory it does `path.dirname` on that directory - and that's because it is working even if we change default folder name. To be honest I didn't know that, anyway nice fun fact 😄

File: packages/cli-doctor/src/tools/healthchecks/xcodeEnv.ts [hunk 1]
```diff
@@ -30,10 +29,19 @@ export default {
...
     try {
       const projectRoot = config?.root ?? findProjectRoot();
-      const missingXcodeEnvFile = findPodfilePaths(projectRoot).some((p) => {
-        const basePath = path.dirname(p);
-        return !pathHasXcodeEnvFile(basePath);
-      });
+
+      const iosFolderPath = config?.project.ios
+        ? config?.project.ios.sourceDir
+        : path.join(projectRoot, 'ios');
```
Comment (thymikee) [inline packages/cli-doctor/src/tools/healthchecks/xcodeEnv.ts:35]: `sourceDir` is set to $root/ios even if it's not explicitly defined

Comment (szymonrybczak) [reply packages/cli-doctor/src/tools/healthchecks/xcodeEnv.ts:35]: Correct!

File: packages/cli-doctor/src/tools/healthchecks/xcodeEnv.ts [hunk 2]
```diff
@@ -29,11 +28,18 @@ export default {
...
   getDiagnostics: async (_, config) => {
     try {
-      const projectRoot = config?.root ?? findProjectRoot();
-      const missingXcodeEnvFile = findPodfilePaths(projectRoot).some((p) => {
-        const basePath = path.dirname(p);
-        return !pathHasXcodeEnvFile(basePath);
-      });
+      const iosFolderPath = config?.project.ios
+        ? config?.project.ios.sourceDir
+        : '';
```
[the remark this answers is not in the document]
Comment (szymonrybczak) [reply packages/cli-doctor/src/tools/healthchecks/xcodeEnv.ts]: Correct, I don't know why I overcomplicated this so much 🫣

**Comment by thymikee** [comment]
cc @arushikesarwani94

**Review by arushikesarwani94 — COMMENTED** [review]
LGTM !

**Pull request merged**